### PR TITLE
Add helpful and harmless example

### PR DESCRIPTION
**Pull request by reciprocated**
This PR adds an example of training on Anthropic's [dataset](https://github.com/anthropics/hh-rlhf)

PPO: https://wandb.ai/sorry/trlx/runs/ab0ehsxm
SFT: https://wandb.ai/sorry/trlx/runs/0aulb18k
ILQL: https://wandb.ai/sorry/trlx/runs/30ab7epk

### CODE REVIEW

File: examples/hh/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+As an example, the following setup assumes a single machine with 8xA100 80GB, the last of which will be dedicated to hosting a reward model. Optionally you can use [Triton Inference Server](https://github.com/triton-inference-server) to host it elsewhere, otherwise the training script will instantiate it (a default one) on its own.
+
+Launch training on 7 GPUs with 8th GPU hosting a reward model:
+```sh
+accelerate launch --num_processes 7 --config_file ../../configs/deepspeed/zero2-bf16.yaml ppo_hh.py
+```
+
+#### Optional steps to setup a reward model (trained with [Dahoas/reward-modeling](https://github.com/Dahoas/reward-modeling)) with Triton Server:
+
```
Comment (Dahoas) [inline examples/hh/README.md:11]: Add line about `pip install tritonclient`

File: examples/hh/configs/ppo_hh.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  checkpoint_interval: 10000
+  eval_interval: 500
+
+  pipeline: "PromptPipeline"
+  trainer: "AcceleratePPOTrainer"
+  checkpoint_dir: "checkpoints/ppo_hh"
+
+model:
+  model_path: "EleutherAI/gpt-j-6B"
```
Comment (Dahoas) [inline examples/hh/configs/ppo_hh.yml:15]: Maybe let's make the default example use a smaller model so more people can run it

**Review by jon-tow — CHANGES_REQUESTED** [review]
Thanks, Max! I left a tiny nit for the `README.md`. 

On another note, I'm hitting an issue when using the GPU-hosting launch command from the `README.md` (for both ILQL and PPO) where the samples list provided to the ppo_hh `reward_fn` is empty causing an `IndexError`. Could you please try this? 

- Traceback Info: https://gist.github.com/jon-tow/e8aefbd0e7b06a1afc21dba38238d089

Note this only occurs when `--num_processes` is provided an odd number (quickly checked for 2, 4, and 6 processes fine).

**Comment by reciprocated** [comment]
@jon-tow regarding your issue, If you upgrade to `accelerate==0.16.0` it will be resolved (I could reproduce it with `accelerate==0.15.0`)

**Review by jon-tow — APPROVED** [review]
@reciprocated I had forgotten about the `accelerate==0.15.0` metric bug 😅 This looks great from my end (not sure if Alex has any more feedback).

**Pull request merged**